android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiToolbar.java

```diff
@@ -0,0 +1,606 @@
+	private final int TOOLBAR_HIDE_OVERFLOW_MENU = 10005;
+	private final int TOOLBAR_SET_LOGO = 10006;
+	private final int TOOLBAR_SET_NAVIGATION_ICON = 10007;
+	private final int TOOLBAR_SET_OVERFLOW_MENU_ICON = 10008;
+	private final int TOOLBAR_DISMISS_POPUP_MENUS = 10009;
+	private final int TOOLBAR_SET_TITLE = 10010;
+	private final int TOOLBAR_SET_TITLE_TEXT_COLOR = 10011;
+	private final int TOOLBAR_SET_SUBTITLE = 10012;
+	private final int TOOLBAR_SET_SUBTITLE_TEXT_COLOR = 10013;
+	private final int TOOLBAR_COLLAPSE_ACTION_VEIW = 10014;
+	private final int TOOLBAR_SET_CONTENT_INSET_END_WITH_ACTIONS = 10015;
+	private final int TOOLBAR_SET_CONTENT_INSET_START_WITH_NAVIGATION = 10016;
+	private final int TOOLBAR_SET_CONTENT_INSETS_ABSOLUTE = 10017;
+	private final int TOOLBAR_SET_CONTENT_INSETS_RELATIVE = 10018;
+	//endregion
+
+	/**
+	 * Constructs a TiUIView object with the associated proxy.
+	 * @param proxy the associated proxy.
+	 * @module.api
+	 */
+	public TiToolbar(TiViewProxy proxy) {
+		super(proxy);
+		toolbar = new Toolbar(proxy.getActivity());
+		toolbar.setContentInsetsAbsolute(0,0);
+		setNativeView(toolbar);
+	}
+
+	/**
+	 * Adds custom views in the toolbar
+	 * @param proxies View proxies to be used
+	 */
+	public void setItems(TiViewProxy[] proxies) {
+		if (proxies != null) {
+			for (int i = 0; i < proxies.length; i++) {
+				toolbar.addView(convertLayoutParamsForView(proxies[i].getOrCreateView()));
+			}
+		}
+	}
+
+	/**
+	 * Sets the background color of the toolbar
+	 * @param color String in Titanium color format
+	 */
+	public void setToolbarColor(String color) {
+		if (!TiApplication.isUIThread()) {
+			TiMessenger.sendBlockingMainMessage(mainHandler.obtainMessage(TOOLBAR_SET_COLOR), color);
+		} else {
+			handleBackgroundColor(color);
+		}
+	}
+
+	/**
+	 * Handler for background color change.
+	 * @param color String in Titanium color format
+	 */
+	private void handleBackgroundColor(String color) {
+		toolbar.setBackgroundColor((TiColorHelper.parseColor(color)));
+		if (((Boolean) proxy.getProperty(TiC.PROPERTY_TRANSLUCENT))) {
+			toolbar.getBackground().setAlpha(BACKGROUND_TRANSLUCENT_VALUE);
+		}
+	}
+
+	/**
+	 * Sets up the Toolbar to extend behind the Status Bar.
+	 * This is useful when the Toolbar instance is set as a support bar in an activity
+	 * and it positioned at the very top of it.
+	 * @param value Boolean value to set. True makes the Toolbar's background extend. False - default behaviour.
+	 */
+	public void setToolbarExtendBackground(boolean value) {
+		if (!TiApplication.isUIThread()) {
+			TiMessenger.sendBlockingMainMessage(mainHandler.obtainMessage(TOOLBAR_SET_BACKGROUND_EXTENDED), value);
+		} else {
+			handleBackgroundExtended(value);
+		}
+	}
+
+	/**
+	 * Handler for extending the background.
+	 * @param value Boolean value to set. True makes the Toolbar's background extend. False - default behaviour.
+	 */
+	private void handleBackgroundExtended(boolean value) {
+		if (value) {
```

Comment:
Calling `handleBackgroundExtended(false)` will do nothing. So, once you've set it to true, it's locked-in forever. This was the intent, right?

If so, then I would get rid of the "value" argument.